### PR TITLE
`tempfile.pyi`: readability improvements

### PR DESCRIPTION
**Pull request by AlexWaygood**
The name `_DirT` implies that it is a `TypeVar`, but it isn't; it's a generic `TypeAlias` that can be parameterised with a `TypeVar`. Rename it to just `_Dir` instead.

Also, use aliases instead of repeating incredibly long `Literal` annotations six times.

### CODE REVIEW

**Comment by JelleZijlstra** [comment]
Does this work as expected? I recall mypy has issues with types of the form `AnySttr | PathLike[AnyStr]`.

**Comment by AlexWaygood** [comment]
> Does this work as expected? I recall mypy has issues with types of the form `AnyStr | PathLike[AnyStr]`.

I ran mypy on the following test script:

```python
from pathlib import Path
from tempfile import NamedTemporaryFile, mkstemp

class BytesPathLike:
    def __fspath__(self) -> bytes: ...

b: BytesPathLike

# These should all pass
NamedTemporaryFile('r', 2, 'utf-8', None, None, 'foo', 'bar')
NamedTemporaryFile('r', 2, 'utf-8', None, None, 'foo', Path('bar'))
NamedTemporaryFile('r', 2, 'utf-8', None, None, b'foo', b'bar')
NamedTemporaryFile('r', 2, 'utf-8', None, None, b'foo', b)
mkstemp(prefix='foo', dir='bar')
mkstemp(prefix=b'foo', dir=b'bar')

# These should fail
NamedTemporaryFile('r', 2, 'utf-8', None, None, b'foo', 'bar')
NamedTemporaryFile('r', 2, 'utf-8', None, None, 'foo', b)
mkstemp(prefix='foo', dir=b'bar')
mkstemp(prefix=b'foo', dir='bar')
```

The expected successes all succeeded, and the expected failures all failed.

---

Now that I look at the file, there are some _other_ bugs in the file -- in this overload, for example, the `AnyStr` `TypeVar` can go unsolved:

https://github.com/python/typeshed/blob/466f9c2ad7bb65702b268081bcc753778f43f6a0/stdlib/tempfile.pyi#L37-L49

But I'd rather leave fixing that to a separate PR, and keep this one as a simple code-readability enhancement.

**Comment by AlexWaygood** [comment]
> But I'd rather leave fixing that to a separate PR, and keep this one as a simple code-readability enhancement.

I changed my mind, and decided I'd also fix them in this PR.

**Comment by AlexWaygood** [comment]
> Now that I look at the file, there are some other bugs in the file -- in this overload, for example, the `AnyStr` `TypeVar` can go unsolved:

Wait, actually, does this matter at all? The return type is the same either way, so maybe it doesn't matter if the `AnyStr` `TypeVar` here is technically "unsolved" in some situations.

**Comment by AlexWaygood** [comment]
> Maybe I should revert the last commit?

I reverted it.

**Review by srittau — APPROVED** [review]

File: stdlib/tempfile.pyi [hunk 1]
```diff
@@ -31,31 +31,33 @@ TMP_MAX: int
 tempdir: str | None
 template: str
 
-_DirT: TypeAlias = AnyStr | os.PathLike[AnyStr]
+_Dir: TypeAlias = AnyStr | os.PathLike[AnyStr]
```
[the remark this answers is not in the document]
Comment (AlexWaygood) [reply stdlib/tempfile.pyi:34]: I like the idea! I'll see if I can file a PR

**Pull request merged**